data-prepper-plugins/personalize-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/personalize/configuration/PersonalizeSinkConfiguration.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.personalize.configuration;
+
+import jakarta.validation.GroupSequence;
+import org.opensearch.dataprepper.plugins.sink.personalize.dataset.DatasetTypeOptions;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.AssertTrue;
+import software.amazon.awssdk.arns.Arn;
+
+import java.util.List;
+import java.util.Optional;
+
+interface PersonalizeAdvancedValidation {}
+
+/**
+ * personalize sink configuration class contains properties, used to read yaml configuration.
+ */
+@GroupSequence({PersonalizeSinkConfiguration.class, PersonalizeAdvancedValidation.class})
+public class PersonalizeSinkConfiguration {
+    private static final int DEFAULT_RETRIES = 10;
+    private static final String AWS_PERSONALIZE = "personalize";
+    private static final String AWS_PERSONALIZE_DATASET = "dataset";
+    private static final List<DatasetTypeOptions> DATASET_ARN_REQUIRED_LIST = List.of(DatasetTypeOptions.USERS, DatasetTypeOptions.ITEMS);
+
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+    @JsonProperty("dataset_type")
+    @NotNull
+    @Valid
+    private DatasetTypeOptions datasetType;
+
+    @JsonProperty("dataset_arn")
+    private String datasetArn;
+
+    @JsonProperty("tracking_id")
+    private String trackingId;
+
+    @JsonProperty("document_root_key")
+    private String documentRootKey;
+
+    @JsonProperty("max_retries")
+    private int maxRetries = DEFAULT_RETRIES;
+
+    @AssertTrue(message = "A dataset arn is required for items and users datasets.", groups = PersonalizeAdvancedValidation.class)
+    boolean isDatasetArnProvidedWhenNeeded() {
+        if (DATASET_ARN_REQUIRED_LIST.contains(datasetType)) {
+            return datasetArn != null;
+        }
+        return true;
+    }
+
+    @AssertTrue(message = "dataset_arn must be a Personalize Dataset arn", groups = PersonalizeAdvancedValidation.class)
+    boolean isValidDatasetArn() {
+        if (datasetArn == null) {
+            return true;
+        }
+        final Arn arn = getArn();
+        boolean status = true;
+        if (!AWS_PERSONALIZE.equals(arn.service())) {
+            status = false;
+        }
+        final Optional<String> resourceType = arn.resource().resourceType();
```

Comment:
as mentioned above, may be good idea to validate accountId, wdyt?

Reply:
Not too concerned with validating accountId here since we have that validation for the Put* APIs. We can't validate accountId for interactions data here since we would only have the trackingId